fastplotlib/widgets/image.py

```diff
@@ -856,7 +865,161 @@ def show(self):
         VBox
             ``ipywidgets.VBox`` stacking the plotter and sliders in a vertical layout
         """
-        # start render loop
-        self.plot.show()
+        canvas = self.plot.show(toolbar=False)
+
+        # check if in jupyter notebook or not
+        if not isinstance(canvas, JupyterWgpuCanvas):
+            return VBox([self.plot.show(toolbar=False), self.widget])
+
+        if toolbar and self.toolbar is None:
+            self.toolbar = ImageWidgetToolbar(self).widget
+            return VBox([canvas, self.toolbar, self.widget])
+        elif toolbar and self.toolbar is not None:
+            return VBox([canvas, self.toolbar, self.widget])
+        else:
+            return VBox([canvas, self.widget])
+
+
+class ImageWidgetToolbar:
```

Comment:
most of the stuff in here is not necessary, you can just get the toolbar from the `Plot` or `GridPlot` instance.